service/header/store_heightsub.go

```diff
@@ -40,7 +40,7 @@ func (hs *heightSub) SetHeight(height uint64) {
 // It can return errElapsedHeight, which means a requested header was already provided
 // and caller should get it elsewhere.
 func (hs *heightSub) Sub(ctx context.Context, height uint64) (*ExtendedHeader, error) {
-	if hs.Height() >= height || hs.Height() == 0 { // if height is zero - intentionally avoid subscribing
```

Comment:
I don;t know how but this if statement is not letting the light node sub to headers in the specific test 

cc: @Wondertan